File: src/main/java/popa/catalin/lab3/command/AttackCommand.java
package popa.catalin.lab3.command;

import popa.catalin.lab3.pieces.ChessPiece;
import popa.catalin.lab3.Position;

public class AttackCommand {
    private static Position invalidPosition = new Position(-1, -1);

    public static void attack(ChessPiece attackingPiece, ChessPiece attackedPiece) {
        Position newPosition = attackedPiece.getPosition();
        attackingPiece.setPosition(newPosition);

        attackedPiece.setAlive(false);
        attackedPiece.setPosition(invalidPosition);
    }
}

File: LabsSimonaBurlacu232/Lab3/ChessGame/ChessGame/Command/AttackCommand.cs
using ChessGame.Pieces;

namespace ChessGame.Command
{
    public class AttackCommand
    {
        public void ExecuteCommnad(ChessPiece from, ChessPiece to)
        {
            from.Position.X = to.Position.X;
            from.Position.Y = to.Position.Y;
            to.IsAlive = false;
        }
    }
}
